src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/PSRecoveryServicesClient/PSRecoveryServicesClient.cs

```diff
@@ -259,6 +261,44 @@ private SiteRecoveryManagementClient GetSiteRecoveryClient()
 
             return siteRecoveryClient;
         }
+
+        /// <summary>
+        /// Gets Vault Upgrade client.
+        /// </summary>
+        /// <param name="location">Resource location.</param>
+        /// <param name="resourceName">Resource Name.</param>
+        /// <param name="resourceType">Resource Type.</param>
+        /// <returns>Vault Upgrade Management client.</returns>
+        private RecoveryServicesVaultUpgradeManagementClient GetVaultUpgradeClient(
+            string location,
+            string resourceName,
+            string resourceType)
+        {
+            string resourceNamespace = Environment.GetEnvironmentVariable("RESOURCE_NAMESPACE");
```

Comment:
@neetus15 I think this was mentioned in the review to Praveen, if you need to pass in some custom object/data, you can use HttpHandler and pass it to ServiceClient. All classes are partial, so you should be able to extend any of the classes
Extend you service client and pass in an HttpHandler that will setup you custom value in your request.
Please understand, you cannot start adding test hooks because it's easy to implement.
Bottom line, are you guys comfortable with this implementation?
If you still think you want to go ahead with this implementation, please open an issue that you would be fixing this and the time frame when this will be fixed.
We can make exception, but we need a plan as to how you will be fixing it.